packages/frontend/electron/src/main/shared-storage/json-file.ts

```diff
@@ -0,0 +1,126 @@
+import fs from 'node:fs';
+
+import type { Memento } from '@toeverything/infra';
+import {
+  backoffRetry,
+  effect,
+  exhaustMapWithTrailing,
+  fromPromise,
+} from '@toeverything/infra';
+import { debounceTime, Observable, timeout } from 'rxjs';
+
+export class PersistentJSONFileStorage implements Memento {
+  data: Record<string, any> = {};
+  subscriptions: Map<string, Set<(p: any) => void>> = new Map();
+  subscriptionAll: Set<(p: Record<string, any>) => void> = new Set();
+
+  constructor(readonly filepath: string) {
+    try {
+      this.data = JSON.parse(fs.readFileSync(filepath, 'utf-8'));
```

Comment:
better check if file exists before reading. no need to throw if it is not being created yet.

Reply:
![CleanShot 2024-07-15 at 10 26 40](https://github.com/user-attachments/assets/7927af06-e584-417c-b672-edfa49e91a0a)

It is best practice to read the file directly.